src/System.Management.Automation/utils/unicode/CaseFolding-g2.cs

```diff
@@ -0,0 +1,356 @@
+namespace System.Management.Automation.Unicode
+{
+    /// <summary>
+    /// Simple case folding methods.
+    /// </summary>
+    internal static partial class SimpleCaseFolding
+    {
+        private static readonly ushort[] L1 =
+        {
```

Comment:
Can you add some comments about what L1 and L2 represent?

Reply:
An idea is to use an array for mapping (simple case folding) a source char to target char. A problem is that SimpleCase.txt contains very small account (1.5Kb) of Unicode code points from Plane0 (SMP) and Plane1 - it is 128 Kb that's a lot for the mapping array. In related Dotnet CoreFXlab issue https://github.com/dotnet/corefxlab/issues/2610 @tarekgh suggested to use the 3-level (8-4-4) mapping technique (and published a gen.cs code in gist). It is ~6Kb. This turned out to be much slower than a single-level array and I tried a two-level array. It is ~10Kb. 
Yesterday I  found that I can use 1-level mapping for chars < 0x5ff and 2-level mapping for other chars. It is ~10Kb too and fastest case I have found. I'll push the commit today.
I guess the current code still has a lot of bugs and I consider it as "alfa". Today I plan to fix surrogates.